wiki/Operator-101.md

```diff
@@ -0,0 +1,124 @@
+Applying manifest on cluster will create all required resources to deploy successfully Contrail and status of all components is available to check at any time of deployment as well as during lifetime of a cluster.
+
+Contrail may be configured in a various ways of configuration for various infrastructures.<br/>
+Example infrastructure would be below cluster with 3 master nodes and 3 worker nodes.
+
+![operator-contrail-scheme](images/operator-contrail-scheme.png)
+
+Master node contains Contrail controller components.<br/>
+Rabbitmq, Zookeeper and Cassandra create database with configuration.<br/>
+From configuration database Config components takes or stores current configuration of cluster.<br/>
+Configuration may be set with WebUI and it's read and propagated by Control components to vRouters which manage networking on an interface on remote machines that are managed by Contrail.<br/>
+Controller also contains Kubemanager component which gathers information from Kubernetes cluster and provides it to configuration.<br/>
+vRouters mounted on every node control networking traffic on an interface with rules defined by agent that gethers rules from control node.<br/>
+Controller components may also have more than one replica spread across multiple nodes which make Contrail work in HA mode.<br/>
+
+In order to deploy such infrastructure on Kubernetes cluster it's necessary to create manifest which defines Manager custom resource.<br/>
+Manager is a single resource which defines Contrail cluster.<br/>
+In it's definition user defines which components should be created and with what configuration.
+For reference on example Manager manifest which implements infrastructure showed above refer to [this](https://github.com/Juniper/contrail-operator/blob/master/deploy/openshift/manifests/0000000-contrail-09-manager.yaml) file.
+
+However, Kubernetes cluster by default does not know what is Manager resource how to implement it's deployment.<br/>
+Because of that it's necessary to beforewards apply all CRDs (Custom Resource Definitions) to cluster which in this repository are located under [*deploy/crds*](https://github.com/Juniper/contrail-operator/tree/master/deploy/crds) directory.<br/>
+While Kubernetes cluster will now properly read manifests for Contrail custom resources, it does not have logic that should be used in order to properly controll resources.
+
+
+
+To fix that problem, operator itself has to be deployed on cluster.<br/>
+Example manifest may be found [here](https://github.com/Juniper/contrail-operator/blob/master/deploy/openshift/manifests/0000000-contrail-08-operator.yaml).
+Applied operator will create separate Pod which will act as controller of custom resources in cluster.
+
+Afterwards, when manifest is applied on cluster in namespace *contrail* (*contrail* namespace has to be created beforehand) may be observed status of all pods created by Contrail.<br/>
```

Comment:
in the *contrail* namespace (...), status of all pods create by Contrail may be observed.